src/General/Binary.hs

```diff
@@ -0,0 +1,231 @@
+{-# LANGUAGE FlexibleInstances, ScopedTypeVariables, Rank2Types #-}
+
+module General.Binary(
```

Comment:
I think you can get rid of this whole module, and avoid providing the instances in FileName/FileInfo - it's unnecessary if you are using cereal.